src/app/product-details-page/components/product-details-page/related-products/related-products.component.html

```diff
@@ -0,0 +1,8 @@
+<div *ngIf="products?.length">
```

Comment:
Move this check to the parent component and just don't render the child one. Child component should definitely have the product and doesn't validate it's absence